cluster/src/main/scala/io/snappydata/ToolsCallbackImpl.scala

```diff
@@ -19,42 +19,39 @@ package io.snappydata
 import java.io.File
 
 import org.apache.spark.SparkContext
-import org.apache.spark.sql.catalyst.expressions.{Attribute, Expression}
-import org.apache.spark.sql.catalyst.plans.physical.{OrderlessHashPartitioning, Partitioning}
+import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.catalyst.plans.physical.{HashPartitioning, Partitioning}
 import org.apache.spark.ui.SnappyDashboardTab
 import org.apache.spark.util.SnappyUtils
 
 object ToolsCallbackImpl extends ToolsCallback {
 
-  def getOrderlessHashPartitioning(partitionColumns: Seq[Expression],
-      partitionColumnAliases: Seq[Seq[Attribute]],
-      numPartitions: Int, numBuckets: Int, tableBuckets: Int): Partitioning = {
-    OrderlessHashPartitioning(partitionColumns, partitionColumnAliases,
-      numPartitions, numBuckets, tableBuckets)
+  def getHashPartitioning(partitionColumns: Seq[Expression],
+      numPartitions: Int, numBuckets: Int): Partitioning = {
+    HashPartitioning(partitionColumns, numPartitions, numBuckets)
   }
 
-  override def checkOrderlessHashPartitioning(partitioning: Partitioning): Option[
-      (Seq[Expression], Seq[Seq[Attribute]], Int, Int, Int)] = partitioning match {
-    case p: OrderlessHashPartitioning => Some(p.expressions, p.aliases,
-      p.numPartitions, p.numBuckets, p.tableBuckets)
+  override def checkHashPartitioning(partitioning: Partitioning): Option[
+      (Seq[Expression], Int, Int)] = partitioning match {
+    case p: HashPartitioning => Some(p.expressions, p.numPartitions, p.numBuckets)
     case _ => None
   }
```

Comment:
Same question. What does buckets here change? Since this won't work in smart connector so what is the difference in behaviour?

Reply:
Yes, In connector mode the scan will happen bucket wise always. When scan starts based on "linkBucketsToPartitions" flag it will determine number of scan partitions. This functionality can also be taken up at optimizer level, so that it works for both connector and embedded mode.